lib/jwt_signed_request/sign.rb

```diff
@@ -38,12 +40,24 @@ def call
 
     private
 
-    attr_reader \
-      :method, :path, :body, :headers,
-      :key_id, :lookup_key_id, :issuer, :additional_headers_to_sign
+    attr_reader(
+      :method,
+      :path,
+      :body,
+      :headers,
+      :key_id,
+      :lookup_key_id,
+      :issuer,
+      :additional_headers_to_sign,
+      :key_store_id,
+    )
 
     def stored_key
-      @stored_key ||= JWTSignedRequest.key_store.get_signing_key(key_id: lookup_key_id)
+      @stored_key ||= key_store.get_signing_key(key_id: lookup_key_id)
+    end
+
+    def key_store
+      key_store_id.nil? ? JWTSignedRequest.key_store : JWTSignedRequest.key_store(key_store_id)
```

Comment:
This logic is duplicated in both this `Sign` class and `Verify`. Consider moving this complexity into the `JWTSignedRequest.key_store` method.

Reply:
Updated: be3659c